Document/0x04g-Testing-Cryptography.md

```diff
@@ -130,6 +134,14 @@ Through source code analysis the following non-exhausting configuration options
 - Fit-for-purpose block encryption modes
 - Key management being done properly
 
+#### Dynamic Analysis
+
+Various weaknesses can be detected within out access to the source code:
+
+- if access to encrypted files or to generated hashes is possible (e.g., through stored files) the same file can be encrypted by two users. If this results in the same encrypted file, a static encryption key has been used.
+- if two different users are able to generate the same hash for the same file, then no (or an indequate) salt has been used.
```

Comment:
Good question. I would keep them separated as they attack different cryptographic directives but can change them if you want to.